File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        d = {}
        for i in arr:
            if d.get(i) is not None:
                d[i] += 1
            else:
                d[i] = 1
        
        s = set(list(d.values()))
        
        return len(s) == len(d.values())
        

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        map<int, int>mp;
        for (int i=0;i<arr.size();i++){
            if (mp.find(arr[i]) != mp.end()) {
                mp[arr[i]] += 1;
            }
            else {
                mp[arr[i]] = 1;
            }
        }
        
        set<int>s;
        map<int,int>::iterator it;
        for (it = mp.begin(); it != mp.end(); it++){
            s.insert(it->second);
        }
        
        return s.size() == mp.size();
    }
};